addons/mail/models/mail_thread.py

```diff
@@ -4757,7 +4776,7 @@ def _get_allowed_access_params(self):
         return set()
 
     @api.model
-    def _get_thread_with_access(self, thread_id, mode="read", **kwargs):
+    def _get_thread_with_access(self, thread_id: int, *, mode="read", **kwargs):
```

Comment:
We don't type; moreover useless (thread_id should be an int, otherwise you should rename the parameter).